src/cascadia/TerminalApp/CommandPalette.xaml

```diff
@@ -36,6 +36,151 @@ the MIT License. See LICENSE in the project root for license information. -->
+                                    Padding="2,0,2,0"
+                                    HorizontalAlignment="Right"
+                                    VerticalAlignment="Center">
+
+                            <TextBlock
+                                        Style="{ThemeResource KeyChordTextBlockStyle}"
+                                        FontSize="12"
+                                        Text="{x:Bind Item.KeyChordText, Mode=OneWay}" />
+                        </Border>
+
+                        <!-- xE70E is ChevronUp. Rotated 90 degrees, it's _ChevronRight_ -->
+                        <FontIcon
+                                    FontFamily="Segoe MDL2 Assets"
+                                    Glyph="&#xE70E;"
+                                    HorizontalAlignment="Right"
+                                    Visibility="{x:Bind Item,
+                                                 Mode=OneWay,
+                                                 Converter={StaticResource HasNestedCommandsVisibilityConverter}}"
+                                    Grid.Column="2">
+
+                            <FontIcon.RenderTransform>
+                                <RotateTransform CenterX="0.5" CenterY="0.5" Angle="90"/>
+                            </FontIcon.RenderTransform>
+                        </FontIcon>
+
+                    </Grid>
+                </ListViewItem>
+            </DataTemplate>
+
+            <DataTemplate x:Key="TabItemTemplate" x:DataType="local:FilteredCommand">
+
+                <!-- This HorizontalContentAlignment="Stretch" is important
+                        to make sure it takes the entire width of the line -->
+                <ListViewItem HorizontalContentAlignment="Stretch"
+                                  IsTabStop="False"
+                                  AutomationProperties.Name="{x:Bind Item.Name, Mode=OneWay}"
+                                  AutomationProperties.AcceleratorKey="{x:Bind Item.KeyChordText, Mode=OneWay}">
+
+                    <Grid HorizontalAlignment="Stretch" ColumnSpacing="8" >
+                        <Grid.ColumnDefinitions>
+                            <ColumnDefinition Width="16"/>
+                            <!-- icon / progress -->
+                            <ColumnDefinition Width="Auto"/>
+                            <!-- command label -->
+                            <ColumnDefinition Width="*"/>
+                            <!-- Spanner -->
+                            <ColumnDefinition Width="Auto"/>
+                            <!-- Indicators -->
+                            <ColumnDefinition Width="16"/>
+                            <!-- gutter for scrollbar -->
+                        </Grid.ColumnDefinitions>
+
+                        <mux:ProgressRing
+                            Grid.Column="0"
+                            IsActive="{x:Bind Item.(local:TabPaletteItem.IsProgressRingActive), Mode=OneWay}"
```

Comment:
Can you make the `DataType` for the `DataTemplate` `local:TabPaletteItem` to get around all this casting? Or does that cause issues with the `PaletteItemTemplateSelector`?

Reply:
I am not sure how to do it - as the data type the template gets is `FilteredCommand`, rather than `TabPaletteItem` (the cast is done on the property)

Reply:
_That feels so weird to me_. But I guess that's just how XAML works haha.